pkg/controllers/job/job_controller_actions.go

```diff
@@ -296,13 +296,17 @@ func (cc *jobcontroller) syncJob(jobInfo *apis.JobInfo, updateStatus state.Updat
 	}
 
 	var jobCondition batch.JobCondition
+	oldStatus := job.Status
 	if !syncTask {
 		if updateStatus != nil {
-			if updateStatus(&job.Status) {
```

Comment:
Pleas check what `updateStatus` has done?

Reply:
sorry i didnt get you

Reply:
I don't know if this is the point of view. Can we print out what has been updated so that users can better detect the problem during debugging?

Reply:
> sorry i didnt get you

UpdateStatus has checked weather updating is need.
Now your changes cause the deep equal always is false, right?

Reply:
You can refer to https://github.com/volcano-sh/volcano/pull/3514 @babugeet .

Reply:
@Monokaix , @lowang-bh : updated the changes pls check